### PR TITLE
feat(auth-admin): Delegation ui and gql

### PR DESCRIPTION
**Pull request by magnearun**
## What

Ui for creating a custom delegation
Function for getting a Zendesk ticket by id
Graphql mution for creating a delegation 

## Why

For admin to handle creating a custom delegation when received on paper

## Screenshots / Gifs

Screenshot 2024-09-10 at 14 36 42

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new mutation for creating delegation entities via the GraphQL API.
	- Added a dedicated screen for creating delegations in the admin portal.
	- Implemented a new modal for confirming delegation actions.
	- Enhanced delegation creation capabilities with a new service method.

- **User Experience Improvements**
	- Improved layout and navigation in the admin interface for easier delegation management.
	- Implemented a structured form for capturing delegation details with real-time validation.
	- Enhanced user messaging with localization support in the confirmation modal.

### CODE REVIEW

**Review by GunnlaugurG — COMMENTED** [review]
Great job, minor comments .. One thing this PR is missing tho. According to the design there is also a button to create a new delegation from `'/delegation-admin/:nationalId'`. That button should open this page with the toNationalId preffiled.

File: libs/portals/admin/delegation-admin/src/screens/DelegationAdminDetails/DelegationAdmin.tsx [hunk 1]
```diff
@@ -10,21 +17,51 @@ import React from 'react'
...
+            <Button
+              icon="arrowForward"
+              onClick={() => {
+                const query = new URLSearchParams({
+                  fromNationalId: delegationAdmin.nationalId,
+                })
+
+                navigate(
+                  `${
+                    DelegationAdminPaths.CreateDelegation
```
Comment (GunnlaugurG) [inline libs/portals/admin/delegation-admin/src/screens/DelegationAdminDetails/DelegationAdmin.tsx:54]: ~ We have a function for this called replaceParams

```suggestion
    import { replaceParams } from '@island.is/react-spa/shared'
    
    navigate(
        replaceParams({
            href: DelegationAdminPaths.CreateDelegaion,
            params: { fromNationalId: delegationAdmin.nationalId }
        })
    )
```

File: libs/portals/admin/delegation-admin/src/screens/DelegationAdminDetails/DelegationAdmin.tsx [hunk 2]
```diff
@@ -10,21 +17,51 @@ import React from 'react'
...
+            title={delegationAdmin.name}
+            intro={delegationAdmin.nationalId}
+          />
+        </GridColumn>
+        {hasAdminAccess && (
+          <GridColumn span={['12/12', '4/12']}>
+            <Button
+              icon="arrowForward"
+              onClick={() => {
+                const query = new URLSearchParams({
```
Comment (GunnlaugurG) [inline libs/portals/admin/delegation-admin/src/screens/DelegationAdminDetails/DelegationAdmin.tsx:48]: we should always mask nationalIds in query params

File: libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+  const [searchParams, setSearchParams] = useSearchParams()
+  const submit = useSubmit()
+  const actionData = useActionData() as CreateDelegationResult
+  const [noEndDate, setNoEndDate] = React.useState(true)
+  const [fromIdentity, setFromIdentity] = React.useState<Identity | null>(null)
+  const [toIdentity, setToIdentity] = React.useState<Identity | null>(null)
+  const [validTo, setValidTo] = React.useState<Date | null>(null)
+  const [isConfirmed, setIsConfirmed] = React.useState(false)
+  const [fromNationalId, setFromNationalId] = React.useState(
+    () => searchParams.get('fromNationalId') || '',
```
Comment (GunnlaugurG) [inline libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx:56]: national id should be masked so here you would need to unmask it

**Review by GunnlaugurG — COMMENTED** [review]
Great job, minor comments, mostly nit

File: libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+  )
+
+  const ClearButton = ({
+    onClick,
+    loading,
+  }: {
+    onClick: () => void
+    loading: boolean
+  }) => (
+    <button
```
[the remark this answers is not in the document]
Comment (magnearun) [reply libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx:168]: This is just the close icon

File: libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+  const [toNationalId, setToNationalId] = React.useState('')
+
+  const fromInputRef = React.useRef<HTMLInputElement>(null)
+  const toInputRef = React.useRef<HTMLInputElement>(null)
+  const formRef = React.useRef<HTMLFormElement>(null)
+  const [showConfirmModal, setShowConfirmModal] = useState(false)
+  const { showShadow, pxProps } = useDynamicShadow({ rootMargin: '-112px' })
+  const defaultFromNationalId = searchParams.get('fromNationalId')
+
+  console.log('defaultFromNationalId', defaultFromNationalId)
```
Comment (GunnlaugurG) [inline libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx:69]: Remove console.log

File: libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+    }
+  }, [actionData])
+
+  useEffect(() => {
+    async function getFromNationalId() {
+      const unmaskedNationalId = await unmaskString(
+        defaultFromNationalId ?? '',
+        userInfo?.profile.nationalId ?? '',
+      )
+      console.log('unmaskedNationalId', unmaskedNationalId)
```
Comment (GunnlaugurG) [inline libs/portals/admin/delegation-admin/src/screens/CreateDelegation/CreateDelegation.tsx:93]: remove

**Review by GunnlaugurG — APPROVED** [review]
🚀

**Pull request merged**